frontend/src/cljs/imas_seamap/views.cljs

```diff
@@ -614,96 +630,161 @@
+       [layer-card layer {:active?   (some #{layer} active-layers)
+                          :loading?  (loading-layers layer)
+                          :errors?   (error-layers layer)
+                          :expanded? (expanded-layers layer)
+                          :opacity   (layer-opacities layer)}])]
+    [:div
+     [:label.bp3-label.height-managed
+      "Habitat layer for region statistics (only one active layer may be selected at a time):"
+      [b/popover {:position           b/BOTTOM
+                  :class         "full-width"
+                  :popover-class-name "bp3-minimal"
+                  :content            (reagent/as-element
+                                       [b/menu
+                                        (for [layer (filter #(= :habitat (:category %)) active-layers)]
+                                          ^{:key (:layer_name layer)}
+                                          [b/menu-item {:text     (:name layer)
+                                                        :on-click #(re-frame/dispatch [:map.region-stats/select-habitat layer])}])])}
+       [b/button {:text       (get habitat-layer :name "Select Habitat Layer for statistics...")
+                  :class "bp3-fill bp3-text-overflow-ellipsis"
+                  :intent     (when-not habitat-layer b/INTENT-WARNING)
+                  :right-icon "caret-down"}]]]
+     [:div.bp3-callout.bp3-icon-help.height-managed
+      [:h5 "Hints"]
+      [:p "Choose a management boundary and select a habitat layer for
+    spatial summaries. Note that only visible habitat layers will be
+    available for selection."]
+
+      [:p "Click on a management boundary (on the map) to generate
+    habitat statistics for that region, and to download the subsetted
+    benthic habitat data."]]]]})
 
 (defn thirdparty-layers-panel
-  []
+  [{:keys [layers active-layers loading-layers error-layers expanded-layers layer-opacities]}]
   {:title   "Third-Party Layers"
-   :content "Third-Party Layers (WIP)"})
-
-(def seamap-drawer-panels
-  {:drawer-panel/habitat-layers    habitat-layers-panel
-   :drawer-panel/bathy-layers      bathy-layers-panel
-   :drawer-panel/imagery-layers    imagery-layers-panel
-   :drawer-panel/management-layers management-layers-panel
-   :drawer-panel/thirdparty-layers thirdparty-layers-panel})
-
-(defn seamap-drawer
+   :content
+   [:div.sidebar-tab.height-managed
+    [layer-search-filter]
+    [layer-catalogue layers
+     {:active-layers active-layers
+      :loading-fn    loading-layers
+      :error-fn      error-layers
+      :expanded-fn   expanded-layers
+      :opacity-fn    layer-opacities}]]})
+
+(defn drawer-panel-selection
+  [panel map-layers region-stats]
+  (let [{:keys [panel props]} panel
+        {:keys [groups active-layers loading-layers error-layers expanded-layers layer-opacities]} map-layers
+        {:keys [habitat-layer]} region-stats]
+    (case panel
+      :drawer-panel/layer-panel
```

Comment:
This is fine, really.  I think I might have used a defmulti (dispatching on panel) elsewhere to do similar stuff which is worth considering.

Reply:
I might consider doing this later, considering I haven't really used defmulti at all yet.